src/popup.tsx

```diff
@@ -39,6 +40,11 @@ const Popup = () => {
     chrome.runtime.sendMessage({ result });
   };
 
+  const disableGrouping = () => {
+    setIsOn(!isOn);
```

Comment:
It would be better to use the callback in setState 
setIsOn(isOn => { setStorage(!isOn); return !isOn;})

Reply:
done